modules/client/src/interfaces.ts

```diff
@@ -539,26 +536,31 @@ export type ContractWithdrawParams = [string, string, BigNumber, string];
 
 export type PrepareUninstallationParams = {
   daoAddressOrEns: string;
-  versionTag: VersionTag;
   pluginAddress: string;
   pluginInstallationIndex?: number;
+  uninstallationParams?: any;
```

Comment:
Shouldn't it be a `Uint8Array`? empty by default

Reply:
no because the encoding is done inside the plugin uninstallation using the abi that comes from the ipfs metadata, the other approach would be asume that the user is going to do de encoding outside of the prepareUninstallation and ask for an Uint8Array

Reply:
Not sure I follow. Prepare uninstallation params is used by a developer, and in my view the developer should have used an encoder to produce the bytes to pass to the prepare function.

Do you mean that you prefer us doing this instead? My concern is that by defining `any`, developers can just pass anything and they will get little clue from what is going wrong. If there was a more explicit way, we should consider it... since this is a sensitive flow

Reply:
Yeah its going to be hard to be specific here, because the params of the encoder can be literally anything. I dont have a strong opinion on doing the encoding ourselves or leave it to the user, in one case theey call `prepareUninstallation` with the params and in the other you call `getPlugin`to get the ABI, encode the params and you pass them to `prepareUninstallation`. Let me know how you invision it and i'll implement it

Reply:
If we're reasonably sure that parameters will have to be an array, I would at least type it like that. For the rest, if we delegate the data checking to the encoding library, it's fine :+1: